src/kbmod/region_search.py

```diff
@@ -1,424 +1,659 @@
-            The names of the collection to get the dataset type stats for. If None, use self.collections.
-        dataset_types : `list[str]`
-            The names of the dataset types to get the dataset type stats for. If None, use self.dataset_types.
+        ic : ImageCollection
+            The ImageCollection to filter.
+        timestamps : list of floats
+            List of timestamps to keep.
+        time_sep_s : float, optional
+            The maximum separation in seconds between the timestamps in the ImageCollection and the timestamps to keep.
 
         Returns
         -------
-        vdr_data : `astropy.table.Table`
-            An Astropy Table containing the VDR data and associated URIs and RA/Dec center coordinates.
+        None
         """
-        if not collections:
-            if not self.collections:
-                raise ValueError("No collections specified")
-            collections = self.collections
-
-        if not dataset_types:
-            if not self.dataset_types:
-                raise ValueError("No dataset types specified")
-            dataset_types = self.dataset_types
+        if len(self.ic) < 1:
+            return
+        self.ic.filter_by_mjds(mjds, time_sep_s)
 
-        vdr_dict = {"data_id": [], "region": [], "detector": [], "uri": [], "center_coord": []}
+    def generate_patches(
+        self,
+        arcminutes,
+        overlap_percentage,
+        image_width,
+        image_height,
+        pixel_scale,
+        dec_range=[-90, 90],
+    ):
+        """
+        Generate patches of the sky based on the given parameters.
+        The patches are generated in a RA-Dec aligned grid, with the given overlap percentage determing
+        how much overlap there is between adjacent patches. Note that since we are doing overlap in both
+        dimension, an `overlap_percentage` of 50% will produce 4x the patches of a grid with 0% overlap.
 
-        for dt in dataset_types:
-            refs = self.butler.registry.queryDimensionRecords(
-                "visit_detector_region", datasets=dt, collections=collections
-            )
-            for ref in refs:
-                vdr_dict["data_id"].append(ref.dataId)
-                vdr_dict["region"].append(ref.region)
-                vdr_dict["detector"].append(ref.detector)
-                vdr_dict["center_coord"].append(self.get_center_ra_dec(ref.region))
+        The generated list of patches is stored in the `self.patches` attribute, with each `Patch` object
+        containing the center coordinates, width, height, and image size as well as an ID corresponding
+        to its index in list `self.patches`.
 
-        # Now that we have the initial VDR data ids, we can also fetch the associated URIs
-        vdr_dict["uri"] = self.get_uris(vdr_dict["data_id"])
+        Parameters
+        ----------
+        arcminutes : float
+            The size of the patches in arcminutes.
+        overlap_percentage : float
```

Comment:
Might want to clarify here that the float value should be between 0-100, and not 0-1. Simpletons like me are easily confused :)